### PR TITLE
added support for v3 alpha

### PR DESCRIPTION
**Pull request by legaspielaine**

### CODE REVIEW

**Comment by blankart** [comment]
Let's use `'v3` and `v2` instead of 3 and 2.

File: cypress/support/gutenberg/commands/controls.js [hunk 1]
```diff
@@ -62,9 +62,8 @@ function rangeControlReset( name, options = {} ) {
 
 	beforeAdjust( name, null, options )
 	cy.getBaseControl( name, { isInPopover } )
-		.find( 'button' )
-		.contains( containsRegExp( 'Reset' ) )
-		.click( { force: true } )
+		.find( `button${ Cypress.env( 'STACKABLE_VERSION' ) === 3 ? '[aria-label="Reset"]' : ':contains(Reset)' }` )
```
Comment (blankart) [inline cypress/support/gutenberg/commands/controls.js:65]: Let's not depend on `STACKABLE_VERSION` inside gutenberg commands

File: cypress/support/stackable/helpers/modules.js [hunk 2]
```diff
@@ -122,8 +122,8 @@ export const assertBlockBackground = ( selector, options = {}, assertOptions = {
 	} = options
 
 	const _assertBlockBackground = ( viewport, desktopOnly ) => {
-		cy.collapse( 'Block Background' )
 		cy.toggleStyle( 'Block Background' )
+		cy.collapse( 'Block Background' )
```
Comment (blankart) [inline cypress/support/stackable/helpers/modules.js:126]: Do we still need this change?

File: cypress/support/stackable/util.js [hunk 3]
```diff
@@ -60,11 +60,18 @@ export function changeUnit( unit = '', name = '', isInPopover = false ) {
...
+						selector()
+							.find( 'button', { log: false } )
+							.contains( containsRegExp( unit ) )
+							.click( { force: true, log: false } )
+					}
+				} else if ( Cypress.env( 'STACKABLE_VERSION' ) === 3 ) {
+					if ( $baseControl.find( '.stk-control-label__units' ).length ) {
+						selector().find( '.stk-control-label__units button.is-active' ).click( { force: true } )
+						selector().find( `.stk-control-label__units button[data-value="${ unit }"]` ).click( { force: true } )
+					}
```
Comment (blankart) [inline cypress/support/stackable/util.js:74]: Don't use `else if` here.

Prematurely end the function after the first `if` statement instead. With this, the command will still run even if `STACKABLE_VERSION` is not defined.

File: cypress/support/stackable/util.js [hunk 4]
```diff
@@ -85,22 +92,38 @@ export function changeControlViewport( viewport = 'Desktop', name = '', isInPopo
...
+					selector()
+						.find( '.stk-control-responsive-toggle' )
+						.invoke( 'attr', 'aria-expanded' )
+						.then( ariaExpanded => {
+							if ( ariaExpanded === 'true' ) {
+								selector()
+									.find( `button[aria-label="${ viewport }"]` )
+									.click( { force: true, log: false } )
+							}
+						} )
```
Comment (blankart) [inline cypress/support/stackable/util.js:125]: Don't use `else if` here.

Prematurely end the function after the first `if` statement instead. With this, the command will still run even if `STACKABLE_VERSION` is not defined.

File: cypress/support/stackable/util.js [hunk 5]
```diff
@@ -85,22 +92,38 @@ export function changeControlViewport( viewport = 'Desktop', name = '', isInPopo
...
+							selectViewport()
+						} else if ( ! isActive() ) {
+							selectViewport()
+						}
+					} else if ( hovered ) {
 						selectViewport()
 					}
-				} else if ( hovered ) {
-					selectViewport()
+					cy.wait( 1 )
```
Comment (blankart) [inline cypress/support/stackable/util.js:111]: Is `cy.wait` not needed for v3?

Comment (legaspielaine) [reply cypress/support/stackable/util.js:111]: yes we dont need this for v3

File: cypress/support/stackable/util.js [hunk 6]
```diff
@@ -60,6 +60,13 @@ export function changeUnit( unit = '', name = '', isInPopover = false ) {
 	if ( unit ) {
 		selector()
 			.then( $baseControl => {
+				if ( Cypress.env( 'STACKABLE_VERSION' ) === 3 ) {
+					if ( $baseControl.find( '.stk-control-label__units' ).length ) {
+						selector().find( '.stk-control-label__units button.is-active' ).click( { force: true } )
+						selector().find( `.stk-control-label__units button[data-value="${ unit }"]` ).click( { force: true } )
+					}
```
Comment (blankart) [inline cypress/support/stackable/util.js:67]: end the function here. add `return`

File: cypress/support/stackable/util.js [hunk 7]
```diff
@@ -85,6 +92,22 @@ export function changeControlViewport( viewport = 'Desktop', name = '', isInPopo
...
+						.find( '.stk-control-responsive-toggle' )
+						.invoke( 'attr', 'aria-expanded' )
+						.then( ariaExpanded => {
+							if ( ariaExpanded === 'true' ) {
+								selector()
+									.find( `button[aria-label="${ viewport }"]` )
+									.click( { force: true, log: false } )
+							}
+						} )
+				}
```
Comment (blankart) [inline cypress/support/stackable/util.js:108]: end the function here. add `return`

**Review by blankart — APPROVED** [review]

**Pull request merged**